core/src/main/scala/com/mesosphere/usi/core/logic/MesosEventsLogic.scala

```diff
@@ -121,28 +114,34 @@ private[core] class MesosEventsLogic(mesosCallFactory: MesosCalls, offerMatcher:
 
       case UpdateEvent(taskStatus) =>
         val taskId = TaskId(taskStatus.getTaskId.getValue)
-        val podId = podIdFor(taskId)
-        logger.info(
-          s"Received task status update from taskId $taskId and podId $podId with status ${taskStatus.getState}"
-        )(LoggingArgs("taskId" -> taskId, "podId" -> podId))
-
-        val newStatus = state.podStatuses.get(podId) match {
-          case Some(oldStatus) =>
-            oldStatus.copy(taskStatuses = oldStatus.taskStatuses.updated(taskId, taskStatus))
+        podTaskIdStrategy.unapply(taskId) match {
+          case Some((podId, _)) =>
+            logger.info(
+              s"Received task status update from taskId $taskId and podId $podId with status ${taskStatus.getState}"
+            )(LoggingArgs("taskId" -> taskId, "podId" -> podId))
+
+            val newStatus = state.podStatuses.get(podId) match {
+              case Some(oldStatus) =>
+                oldStatus.copy(taskStatuses = oldStatus.taskStatuses.updated(taskId, taskStatus))
+              case None =>
+                PodStatus(podId, Map(taskId -> taskStatus))
+            }
+
+            SchedulerEvents(
+              stateEvents = List(PodStatusUpdatedEvent(podId, Some(newStatus))),
+              mesosCalls = if (taskStatus.hasUuid) {
+                // frameworks should accept only status updates that have UUID set
+                // http://mesos.apache.org/documentation/latest/scheduler-http-api/#acknowledge
+                List(mesosCallFactory.newAcknowledge(taskStatus.getAgentId, taskStatus.getTaskId, taskStatus.getUuid))
+              } else {
+                Nil
```

Comment:
Do we need a warning or error log here? Or can we safely ignore it?

Reply:
No, this is the expected way for frameworks to handle task status updates; acknowledge only needed if status has a uuid